Tests/SonarScanner.MSBuild.PreProcessor.Test/PreProcessorTests.cs

```diff
@@ -499,25 +391,87 @@ private static void AssertExpectedServerSetting(AnalysisConfig actualConfig, str
         actualProperty.Value.Should().Be(expectedValue, "Unexpected property value. Key: {0}", key);
     }
 
-    private static void AssertDirectoryExists(string path) =>
-        Directory.Exists(path).Should().BeTrue("Expected directory does not exist: {0}", path);
-
-    private static PreProcessor CreatePreProcessor(MockObjectFactory factory) =>
-        new(factory, factory.Logger);
-
     private sealed class TestScope : IDisposable
     {
         public readonly string WorkingDir;
+        public readonly MockObjectFactory Factory;
+        public readonly PreProcessor PreProcessor;
 
         private readonly WorkingDirectoryScope workingDirectory;
+        private readonly TestContext testContext;
 
-        public TestScope(TestContext context)
+        public TestScope(TestContext context, MockObjectFactory factory = null)
         {
+            testContext = context;
             WorkingDir = TestUtils.CreateTestSpecificFolderWithSubPaths(context);
             workingDirectory = new WorkingDirectoryScope(WorkingDir);
+            Factory = factory ?? new MockObjectFactory();
+            PreProcessor = new PreProcessor(Factory, Factory.Logger);
+        }
+
+        public void AssertDirectoriesCreated()
+        {
+            var settings = Factory.ReadSettings();
+            AssertDirectoryExists(settings.AnalysisBaseDirectory);
+            AssertDirectoryExists(settings.SonarConfigDirectory);
+            AssertDirectoryExists(settings.SonarOutputDirectory);
+            // The bootstrapper is responsible for creating the bin directory
+        }
+
+        public async Task<bool> Execute(IEnumerable<string> args = null)
+        {
+            args ??= CreateArgs();
+            return await PreProcessor.Execute(args);
+        }
+
+        public AnalysisConfig AssertAnalysisConfig(int numAnalyzers)
+        {
+            var filePath = Factory.ReadSettings().AnalysisConfigFilePath;
+            Factory.Logger.AssertNoErrorsLogged();
+            Factory.Logger.AssertVerbosity(LoggerVerbosity.Debug);
+
+            AssertConfigFileExists(filePath);
+            var actualConfig = AnalysisConfig.Load(filePath);
+            actualConfig.SonarProjectKey.Should().Be("key", "Unexpected project key");
+            actualConfig.SonarProjectName.Should().Be("name", "Unexpected project name");
+            actualConfig.SonarProjectVersion.Should().Be("1.0", "Unexpected project version");
+            actualConfig.AnalyzersSettings.Should().NotBeNull("Analyzer settings should not be null");
```

Comment:
All of these `because` messages are useless. They are reminiscence of the past `Assert.AreEqual`, before fluent assertions provided code-based messages. Remove them all